src/api/cluster/pke/pkeservice/service.go

```diff
@@ -0,0 +1,269 @@
+	}
+
+	proc := brnProcess(process.Process{
+		Id:     nodeStatus.ProcessID,
+		Type:   "pke-bootstrap",
+		Log:    fmt.Sprintf("%s: %s", nodeStatus.Phase, nodeStatus.Message),
+		Status: Running,
+	}, brn)
+
+	if proc.Id == "" {
+		proc.Id = s.idGenerator.New()
+		proc.StartedAt = nodeStatus.Timestamp
+
+		for _, cp := range clusterProcesses {
+			proc.ParentId = cp.Id
+		}
+	} else {
+		existing, err := s.processes.GetProcess(ctx, proc.Id)
+		if err != nil {
+			return resp, wrapServiceError(errors.WrapIf(err, "failed to get existing process"))
+		}
+
+		if proc.OrgId != existing.OrgId || proc.ResourceId != existing.ResourceId || proc.ResourceType != existing.ResourceType || proc.Type != existing.Type {
+			return resp, newServiceError("invalid process ID")
+		}
+		proc = existing
+	}
+
+	if nodeStatus.Final {
+		proc.Status = nodeStatus.Status
+		proc.FinishedAt = &nodeStatus.Timestamp
+		proc.Log = nodeStatus.Message
+	}
+
+	proc, err = s.processes.LogProcess(ctx, proc)
+	if err != nil {
+		return resp, errors.WrapIf(err, "failed to log process")
+	}
+
+	event := pipeline.ProcessEvent{
+		ProcessId: proc.Id,
+		Type:      fmt.Sprintf("pke-%s", nodeStatus.Phase),
+		Log:       nodeStatus.Message,
+		Status:    nodeStatus.Status,
+		Timestamp: nodeStatus.Timestamp,
+	}
+	_, err = s.processes.LogProcessEvent(ctx, event)
+	if err != nil {
+		return resp, err
+	}
+
+	resp.ProcessID = proc.Id
+
+	if nodeStatus.Final && nodeStatus.Status == Failed {
+		signalValue := newServiceError("%s failed: %s", nodeStatus.Name, nodeStatus.Message)
+		for _, cp := range clusterProcesses {
+			// send signals with best effort
+			_ = s.processes.SignalProcess(ctx, cp.Id, "node-bootstrapped", signalValue)
+		}
+	}
+
+	return resp, err
+}
+
+// Store allows looking up clusters form persistent storage
+type Store interface {
+	// GetCluster returns a generic Cluster.
+	// Returns an error with the NotFound behavior when the cluster cannot be found.
+	GetCluster(ctx context.Context, id uint) (Cluster, error)
+}
+
+// processService provides access to pipeline processes.
+// +testify:mock:testOnly=true
+type processService interface {
+	// LogProcess create a process entry
+	LogProcess(ctx context.Context, proc Process) (process Process, err error)
+
+	// LogProcessEvent create a process event
+	LogProcessEvent(ctx context.Context, proc ProcessEvent) (processEvent ProcessEvent, err error)
+
+	// ListProcesses lists access processes visible for a user.
+	ListProcesses(ctx context.Context, query Process) (processes []Process, err error)
+
+	// GetProcess returns a single process.
+	GetProcess(ctx context.Context, id string) (process Process, err error)
+
+	// CancelProcess cancels a single process.
+	// CancelProcess(ctx context.Context, id string) (err error)
+
+	// SignalProcess sends a signal to a single process.
+	SignalProcess(ctx context.Context, id string, signal string, value interface{}) (err error)
```

Comment:
Should inject a cadence client instead IMHO.

Reply:
the cadence client is injected via an interface to the new process workflow service